lib/helper/Appium.js

```diff
@@ -0,0 +1,839 @@
+    } else
+      throw new Error(`getAllContexts method can be used only with apps`);
+  }
+
+  /**
+   * Retrieve current context
+   * Appium: support Android and iOS
+   */
+  grabContext() {
+    if (this.options.desiredCapabilities.platformName) {
+      return this.browser.context();
+    } else
+      throw new Error(`getContext method can be used only with apps`);
+  }
+
+  /**
+   * Get current device activity.
+   * Appium: support only Android
+   */
+  grabCurrentActivity() {
+    if (this.options.desiredCapabilities.platformName == "Android") {
+      return this.browser.getCurrentDeviceActivity();
+    } else
+      throw new Error(`getCurrentDeviceActivity method can be used only with Android apps`);
+  }
+
+  /**
+   * Get the current geolocation.
+   * Only for browser, firstly you sould setGeoLocation
+   * Appium: support Android and iOS
+   */
+  grabGeoLocation() {
+    if (this.options.desiredCapabilities.platformName) {
+      return this.browser.getGeoLocation();
+    } else
+      throw new Error(`getGeoLocation method can be used only with apps`);
+  }
+
+  /**
+   * Get informations about the current network connection (Data/WIFI/Airplane).
+   * The actual server value will be a number. However WebdriverIO additional
+   * properties to the response object to allow easier assertions.
+   * Appium: support only Android
+   */
+  grabNetworkConnection() {
+    if (this.options.desiredCapabilities.platformName) {
+      return this.browser.getNetworkConnection();
+    } else
+      throw new Error(`getNetworkConnection method can be used only with Android apps`);
+  }
+
+  /**
+   * Get current orientation.
+   * Appium: support Android and iOS
+   */
+  grabOrientation() {
+    if (this.options.desiredCapabilities.platformName) {
+      return this.browser.orientation().then(function(res) {
+        return res.value;
+      });
+    } else
+      throw new Error(`grabOrientation method can be used only with apps`);
+  }
+
+  /**
+   * Get all the currently specified settings
+   * Appium: support Android and iOS
+   */
+  grabSettings() {
+    if (this.options.desiredCapabilities.platformName) {
+      return this.browser.settings();
+    } else
+      throw new Error(`grabSettings method can be used only with apps`);
+  }
+
+  /**
+   * Hide the keyboard.
+   * Appium: support Android and iOS
+   */
+  hideDeviceKeyboard(strategy, key) {
+    if (this.options.desiredCapabilities.platformName) {
+      strategy = strategy || 'tapOutside'
+      return this.browser.hideDeviceKeyboard(strategy, key);
+    } else
+      throw new Error(`hideDeviceKeyboard method can be used only with apps`);
+  }
+
+  /**
+   * Install an app on device.
+   * Appium: support only Android
+   */
+  installApp(path) {
+    if (this.options.desiredCapabilities.platformName == "Android") {
+      return this.browser.installApp(path);
+    } else
+      throw new Error(`installApp method can be used only with Android apps`);
```

Comment:
I think the method `onlyForApps` I wrote above can be improved to handle platform differences as well:

```js
onlyForApps.call(this, arguments.calle.name, 'Android')
```